server/modules/nodeDef/manager/nodeDefManager.js

```diff
@@ -25,21 +22,9 @@ const insertNodeDef = async (user, surveyId, nodeDefParam, client = db) =>
 
 // ======= READ
 
-const fetchNodeDefsBySurveyId = async (surveyId, draft = false, advanced = false, client = db) => {
-  const nodeDefsDB = await NodeDefRepository.fetchNodeDefsBySurveyId(surveyId, draft, advanced, client)
-
-  return R.reduce(
-    (acc, nodeDef) => {
-      // remove draft and unpublished nodeDef
-      if (draft || NodeDef.isPublished(nodeDef)) {
-        acc[NodeDef.getUuid(nodeDef)] = nodeDef
-      }
-
-      return acc
-    },
-    {},
-    nodeDefsDB
-  )
+const fetchNodeDefsBySurveyId = async (surveyId, draft = false, advanced = false, includeDeleted = false, client = db) => {
+  const nodeDefsDb = await NodeDefRepository.fetchNodeDefsBySurveyId(surveyId, draft, advanced, includeDeleted, client)
+  return ObjectUtils.toUuidIndexedObj(nodeDefsDb)
```

Comment:
not possible to get the object indexed by the query?

Reply:
it's a bit complex because the record callback function performs even a deep merge of the props